src/test/java/org/openstreetmap/atlas/geography/SegmentTest.java

```diff
@@ -39,6 +41,14 @@ public void testIntersectionWithSameEnd()
                 SEGMENT_SAME_END_2.reversed().intersects(SEGMENT_SAME_END_1.reversed()));
         Assert.assertTrue("Same End is broken", SEGMENT_SAME_END_1.intersects(SEGMENT_SAME_END_2));
         Assert.assertTrue("Same End is broken", SEGMENT_SAME_END_2.intersects(SEGMENT_SAME_END_1));
+        Assert.assertEquals(INTERSECTION_LOCATION,
+                SEGMENT_SAME_END_1.reversed().intersection(SEGMENT_SAME_END_2.reversed()));
+        Assert.assertEquals(INTERSECTION_LOCATION,
+                SEGMENT_SAME_END_2.reversed().intersection(SEGMENT_SAME_END_1.reversed()));
+        Assert.assertEquals(INTERSECTION_LOCATION,
+                SEGMENT_SAME_END_1.intersection(SEGMENT_SAME_END_2));
+        Assert.assertEquals(INTERSECTION_LOCATION,
+                SEGMENT_SAME_END_2.intersection(SEGMENT_SAME_END_1));
     }
 
     @Test
```

Comment:
Can you add and intersection call to the overflowing example as well? The intersection should be Null Island.